src/phoenix/experimental/callbacks/langchain_tracer.py

```diff
@@ -68,10 +77,55 @@ def _prompt_template(serialized: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
                 yield LLM_PROMPT_TEMPLATE_VARIABLES, kwargs.get("input_variables", [])
                 yield LLM_PROMPT_TEMPLATE_VERSION, "unknown"
                 break
-        except TypeError:
+        except (AttributeError, KeyError, TypeError):
+            continue
+
+
+def _invocation_parameters(extra: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
+    """Yields invocation parameters if present."""
+    for key, value in extra.get("invocation_params", {}).items():
+        yield ".".join([LLM_INVOCATION_PARAMETERS, key]), value
+
+
+def _model_name(extra: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yields model name if present."""
+    for key in ["model_name", "model"]:
+        try:
+            yield LLM_MODEL_NAME, extra["invocation_params"][key]
+            break
+        except (KeyError, TypeError):
             continue
 
 
+def _token_counts(outputs: Dict[str, Any]) -> Iterator[Tuple[str, int]]:
+    """Yields token count information if present."""
+    for attribute_name, key in [
+        (LLM_TOKEN_COUNT_PROMPT, "prompt_tokens"),
+        (LLM_TOKEN_COUNT_COMPLETION, "completion_tokens"),
+        (LLM_TOKEN_COUNT_TOTAL, "total_tokens"),
+    ]:
+        try:
+            yield attribute_name, outputs["llm_output"]["token_usage"][key]
+        except (KeyError, TypeError):
+            continue
+
+
+def _function_calls(outputs: Dict[str, Any]) -> Iterator[Tuple[str, int]]:
+    """Yields function call information if present."""
+    try:
+        function_name = outputs["generations"][0][0]["message"]["kwargs"]["additional_kwargs"][
+            "function_call"
+        ]["name"]
+        arguments = json.loads(
+            outputs["generations"][0][0]["message"]["kwargs"]["additional_kwargs"]["function_call"][
+                "arguments"
+            ]
+        )
+        yield ".".join([LLM_FUNCTION_CALL, function_name]), arguments
+    except (KeyError, IndexError, TypeError):
+        ...
```

Comment:
nit. `...` looks like "under construction"
```suggestion
        pass
```